### PR TITLE
Escape special characters in ref name for update ref

### LINKED ISSUE #2453: Update ref doesn't escape special characters in ref name
The `UpdateRef` function doesn't substitute special characters for their escaped versions within ref names before making a patch request. This consistently leads to invalid paths if the ref has a special character (ex: #).

Planning to fix.

### PR DESCRIPTION
**Pull request by shravan1k**
Closes #2453 

Most of the functions that deal with refs make sure to escape the special characters within the ref's name. The `UpdateRef` function doesn't seem to call the function that does this substitution of special characters with their encodings. This PR solves this problem and should now successfully patch remote refs if the `GetRef` function says the corresponding ref exists.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @shravan1k !
Please add a unit test that demonstrates this change in functionality.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Whups - checked the wrong box.

Thank you, @shravan1k !
Please add a unit test that demonstrates this change in functionality.

**Review by gmlewis — APPROVED** [review]
Beautiful, @shravan1k !  Thank you!
LGTM.

Awaiting second LGTM from any other contributor to this repo before merging.

**Review by marcelosousa — APPROVED** [review]
LGTM 👍

**Comment by gmlewis** [comment]
Thank you, @marcelosousa !
Merging.

**Pull request merged**
